src/modules/uavcan/sensors/baro.cpp

```diff
@@ -84,25 +130,29 @@ int UavcanBarometerBridge::ioctl(struct file *filp, int cmd, unsigned long arg)
 
 void UavcanBarometerBridge::air_data_sub_cb(const uavcan::ReceivedDataStructure<uavcan::equipment::air_data::StaticAirData> &msg)
 {
-	auto report = ::baro_report();
-
+	baro_report report;
+	
 	report.timestamp   = msg.getMonotonicTimestamp().toUSec();
 	report.temperature = msg.static_temperature;
 	report.pressure    = msg.static_pressure / 100.0F;  // Convert to millibar
+	report.error_count = 0;
 
 	/*
 	 * Altitude computation
 	 * Refer to the MS5611 driver for details
 	 */
-	const double T1 = 15.0 + 273.15; // temperature at base height in Kelvin
-	const double a  = -6.5 / 1000;   // temperature gradient in degrees per metre
-	const double g  = 9.80665;       // gravity constant in m/s/s
-	const double R  = 287.05;        // ideal gas constant in J/kg/K
+	const float T1 = 15.0f + 273.15f; // temperature at base height in Kelvin
```

Comment:
Actually I'm not so sure if that applies for each and every of those variables. I did compare the single-float conversion and got resolution artefacts when I wrote the driver 3 years ago. It will require a careful re-evaluation, since the pressure differences are rather small numbers and I first need to trace the calculation logic again to understand if any of those are have an influence, of ir only the final conversion has to be double.